sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -2164,6 +2186,20 @@ def __init__(self, frame):
   def names(self):
     return self._frame._expr.proxy().index.names
 
+  @names.setter
+  def names(self, value):
+    def set_index_names(df):
+      df = df.copy()
+      df.index.names = value
+      return df
```

Comment:
Why the copy here

Reply:
This is to avoid mutating the original `DataFrame`, since PCollection elements are supposed to be immutable.

Reply:
sg